components/UpgradeModal.tsx

```
@@ -0,0 +1,190 @@
+import React from 'react';
+import { Button } from './ui/button';
+import { X, Crown, Zap, Infinity, Check, Loader2 } from 'lucide-react';
```

Biome (2.3.13)

[error] Do not shadow the global "Infinity" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)